Library/DiscUtils.Vdi/DiskStream.cs

```diff
@@ -189,7 +189,7 @@ public override int Read(byte[] buffer, int offset, int count)
                     long filePos = _fileHeader.DataOffset + _fileHeader.BlockExtraSize + blockOffset +
                                    offsetInBlock;
                     _fileStream.Position = filePos;
-                    StreamUtilities.ReadFully(_fileStream, buffer, offset + numRead, toRead);
+                    StreamUtilities.ReadExact(_fileStream, buffer, offset + numRead, toRead);
```

Comment:
Did not throw before, does now. Since the number of bytes read was not tracked, I believe the correct behavior is to throw on a short read.